docs/design/RUC-on-type.md

```diff
@@ -0,0 +1,189 @@
+# RequiresUnreferencedCode on Types
+The RequiresUnreferencedCode being allowed on types its a feature that allows developers to quickly mark entire classes as dangerous for trimming. instead of having to mark every method in the class. This document will go over whats the behavior of certain scenarios when the RequiresUnreferencedCode attribute is placed on a type
+
+## RequiresUnreferencedCode on a type warns on instance constructor or static method access
+Having RequiresUnreferencedCode on the type implies that any instance constructor or static method is considered dangerous for trimming. Following are examples of this behavior:
```

Comment:
Let's be a bit more technical. My description would be:

If `RequiresUnreferencedCode` is present on a class declaration, the behavior of the following members behave as if they were annotated directly:

- Static methods
- Instance constructors
- Static properties
- Static events

Members which are not affected are:

- Nested types
- Instance members